Source/WebInspectorUI/UserInterface/Views/TreeOutline.js

```diff
@@ -316,13 +316,19 @@ WI.TreeOutline = class TreeOutline extends WI.Object
         let child = this.children[childIndex];
         let parent = child.parent;
 
-        if (child.deselect(suppressOnDeselect) && !suppressSelectSibling) {
-            if (child.previousSibling)
-                child.previousSibling.select(true, false);
-            else if (child.nextSibling)
-                child.nextSibling.select(true, false);
+        let currentSelectedElement = this.treeOutline.selectedTreeElement;
+        let childOrDescentdantWasSelected = !!child.selfOrDescendant((descentdant) => descentdant === currentSelectedElement);
+
+        let childWasSelected = child.deselect(suppressOnDeselect);
+
+        if (childOrDescentdantWasSelected && !suppressSelectSibling) {
+            const omitFocus = true;
+            if (childWasSelected && child.previousSibling)
```

Comment:
I kinda feel like we should still select the `child.previousSibling`/`child.nextSibling` even if the `child` itself is not selected, as the `child` is still the "lowest common ancestor" of the selection, so we'd probably want the new selection to be as close to the old selection as possible.